src/VisualStudio/Core/Def/Implementation/Remote/ServiceHubRemoteHostClient.ConnectionManager.cs

```diff
@@ -178,6 +172,8 @@ public void Shutdown()
 
                     _pools.Clear();
                 }
+
+                _hubClient.Dispose();
```

Comment:
intentionally outside of lock?  if so, can you doc why?

Reply:
Calling to code that's not ours. Usually good practice to not hold a lock when doing so. 

In this case it seems like the lock is used to manage the pooled connections, so only operations related to that need to be locked.